src/loader/ident_env.cppo.ml

```diff
@@ -131,11 +131,11 @@ let rec extract_signature_type_items vis items =
       `ClassType (id, obj_id, None, vis'=Hidden, None) :: extract_signature_type_items vis rest
 #endif
 
-    | Sig_typext (id, constr, Text_exception, Exported) :: rest ->
+    | Sig_typext (id, constr, Text_exception, vis') :: rest when vis=vis' ->
       `Exception (id, Some constr.ext_loc)
       :: extract_signature_type_items vis rest
 
-    | Sig_typext (id, constr, _, Exported) :: rest ->
+    | Sig_typext (id, constr, _, vis') :: rest when vis=vis'->
```

Comment:
This looks correct but I would be more confident if this function was written in a way that makes use of exhaustive patterns. I can have a try at it separately.

Reply:
PR: https://github.com/ocaml/odoc/pull/1114